tests/cpp/cell/test_tile_iterator.cu

```diff
@@ -0,0 +1,116 @@
+#include "common/test_utils.hpp"
+#include "types/mod.hpp"
+
+namespace tiledcuda {
+namespace testing {
+
+using namespace cell;
+using namespace cute;
+namespace tl = tile_layout;
+
+namespace {
+
+template <typename Element>
+__device__ void init_value(Element* data, int numel) {
+    for (int i = 0; i < numel; ++i) data[i] = static_cast<Element>(i);
+}
+
+template <typename Iterator>
+__global__ void test_tile_iterator() {
+    using DType = typename Iterator::Tile::DType;
+
+    extern __shared__ __align__(sizeof(double)) unsigned char buf_[];
+    auto* buf = reinterpret_cast<DType*>(buf_);
+    init_value(buf, Iterator::Tile::kNumel);
+
+    typename Iterator::Tile s_tile(buf);
+    printf("Shared tile:\n");
+    s_tile.dump_value();
+
+    Iterator tiles(buf);
+
+    printf("Iterate over rows.\n\n");
+    for (int i = 0; i < Iterator::sc0; ++i) {
+        printf("Iteration-[%d, _]:\n", i);
+        tiles(i, _).to_tile().dump_value();
```

Comment:
Slicing a 2D iterator will return a 2D iterator with one dimension reduced to 1, resulting in a 1D iterator. The `to_tile` function can then flatten the iterator into a large tile.